### ISSUE TITLE
numpy.average not working with units

### ISSUE THREAD

**Issue by MaxNoe** [astropy/astropy#5421]
While `numpy.mean` works fine with `astropy.units.Quantity`, `numpy.average` does not.

``` python
In [1]: import numpy as np

In [2]: import astropy.units as u

In [3]: a = np.linspace(0, 1, 10) * u.meter

In [4]: np.mean(a)
Out[4]: <Quantity 0.5 m>

In [5]: np.average(a)
Out[5]: 0.5

In [6]: b = np.linspace(0, 1, 10) * u.centimeter

In [7]: np.average(a) == np.average(b)
Out[7]: True
```

**Comment by mhvk** [comment]
@MaxNoe - this a problem with numpy, and a fix is in the works, but as it is a change in behaviour, first a deprecation warning is given: https://github.com/numpy/numpy/pull/7433

**Comment by mhvk** [comment]
See https://github.com/numpy/numpy/pull/8290 for the upstream fix; presumably this will land in numpy 1.13.

**Comment by mhvk** [comment]
closing this since the fix is in numpy master and will thus be in numpy 1.13.
